### PR TITLE
Change from land polygons to ocean polygons

### LINKED ISSUE #1982: Switch to ocean polygons
We should switch from preprocessed land polygons on a water colour background to preprocessed water polygons on a land colour background.

This would
- Allow easier customization with hillshading
- Be quicker in inland areas
- Allow more water styling, including outlines of water
- Allow rendering on top of maritime borders, making it possible to have them less prominent
- Makes it easier to render inland areas with no coastline by substituting an empty shapefile

Cross-ref #621 

Issues
- There are no simplified water polygons

cc @joto @imagico

### PR DESCRIPTION
**Pull request by jeisenbe**
Fixes #1982 
Reverts #3065
See similar previous PR #2066 which was reverted due to #2101  
Also see #2507
Related to solving several issues including #621, #1547, #1781, #2013, #2025, #2609, #2688, #3695, and PR #3670

### Changes proposed in this pull request:
- Change from rendering  land polygons on an ocean-color background to rendering ocean polygons on a land-color background
- Remove outline from low-zoom coastlines

## Explanation

It is beneficial to render the oceans (that is, all water outside of the openstreetmap coastline) as polygons, rather than the present situation where the continents and islands are rendered as polygons. While this PR does not make any changes in the layering of land and water, with the introduction of water polygons it will become possible to move the oceans above water lines and landcover, which will be helpful in cases where these will show over the ocean. This will allow solving several issues related to mud flats, mangroves, wetlands and areas of parkland at the coast. It also makes it possible to use different colors for rivers and lakes, if desired. It also makes it easier to render maritime borders less prominently

Previously this same change was made in PR #2066 which had to be reverted due to an issue with the simplified water polygons that was seen on a few specific server setups. However, it appears that this issue may no longer be present with current software versions and the current simplified water polygons available from openstreetmapdata.com; see the discussion in issue #2101. Therefore we are attempting to implement ocean polygons using the current

Unfortunately it isn't possible to render a simplified coastline outline, as in PR #3065 and currently, from the simplified water polygons. This could be done by starting to use the generalized polygon files which are also available at openstreetmapdata.com, however this PR is attempting to make the smallest change possible. If the maintainers and contributors agree that a generalized outline is desirable at z0 to z7, I can easier add this feature now or in a separate PR by using the shapefiles at http://openstreetmapdata.com/data/generalized-coastlines

## Test rendering:

Before:
z3-europe-land-polygons-master

After
z3-europe-ocean-polygons-after

- Rendering is same as current rendering at z8 and higher. The only difference is reverting #3065 which simplified the coastline and added a blue outline at z0 to z7

### CODE REVIEW

**Comment by jeisenbe** [comment]
I've opened issue #3695 so that we can discuss whether the coastline outline and simplification or generalization is desireable at low zoom levels.

**Comment by kocio-pl** [comment]
Do you plan to ask affected users if they still have a problem with rendering ocean polygons?

**Comment by jeisenbe** [comment]
> Do you plan to ask affected users if they still have a problem with
rendering ocean polygons?

I’ve now mentioned this PR and tagged the affected users in #2101 -
hopefully they are still using the github.

**Comment by jeisenbe** [comment]
I've now posted to the dev and talk mailing lists, and added a forum post. Are there any other places where we should mention this change to alert potentially affected users?

**Comment by kocio-pl** [comment]
Sounds to me like enough to reach interested persons.

**Comment by rrzefox** [comment]
I've deployed this on my server. Not all tiles have been completely rerendered yet, but I've not seen any completely broken tiles yet (as I did when this was last attempted).
Apart from the difference due to the outline, I've seen differences here:
http://bl.ocks.org/matthijsmelissen/raw/af7a602c222dbf1ff1a2c0d84ed755b7/#9.00/66.7063/13.0012
I'm not sure if that is to be expected or even related to this PR (might just be the tiles on osm.org being too old).

**Comment by jeisenbe** [comment]
Thank you, @rrzefox for testing this again. It's very helpful to be able to see the whole planet for testing. 
I didn't see any differences at the link which you shared above, other than the newly visible protected areas (due to PR #3661). But perhaps I'm not looking at the right place. Is it visible on z9 here? http://bl.ocks.org/matthijsmelissen/raw/af7a602c222dbf1ff1a2c0d84ed755b7/#9.00/66.7063/13.0012

**Comment by rrzefox** [comment]
The protected areas were what I meant - I simply forgot that osm.org hasn't deployed 4.20 yet.

**Comment by pnorman** [comment]
I'm +1 to the change in principle, but haven't given this an in-depth review.

**Comment by imagico** [comment]
Note it would be good to test this preemptively for the new version of the coastline process now available for evaluation on 

https://osmdata.openstreetmap.de/

This should be backwards compatible but would none the less be good to test.  Same applies if we decide to keep using land polygons. See

https://blog.jochentopf.com/2019-03-07-the-new-osmdata-service.html

**Comment by jeisenbe** [comment]
I’ve pushed a new commit which changes the links for the ocean polygons to
osmdata.openstreetmap.de/download/

I did not change the Antarctic ice sheet files. This can be fixed in a
separate PR, to avoid too much complexity.

However, I haven’t yet had time to download the full shapefiles from the
new source; they are very large.

>

**Comment by imagico** [comment]
Note i did not mean we should change to source the coastline data from osmdata.openstreetmap.de right now in this PR.  I meant we should check if it works this way.  You changing the PR makes this easier for people to do of course.  The eventual move to the new source should be done uniformly for all files and be coordinated with the OSMF sysadmins.

File: scripts/get-shapefiles.py [hunk 1]
```diff
@@ -47,11 +47,11 @@
     },
 
     2: {
-        'directory': 'simplified-land-polygons-complete-3857',
-        'url': 'http://data.openstreetmapdata.com/simplified-land-polygons-complete-3857.zip',  # noqa
+        'directory': 'simplified-water-polygons-complete-3857',
+        'url': 'http://osmdata.openstreetmap.de/download/simplified-water-polygons-complete-3857.zip',  # noqa
```
Comment (imagico) [inline scripts/get-shapefiles.py]: This does not work with `http`, only with `https`.

File: scripts/get-shapefiles.py [hunk 2]
```diff
@@ -63,11 +63,11 @@
     },
 
     4: {
-        'directory': 'land-polygons-split-3857',
-        'url': 'http://data.openstreetmapdata.com/land-polygons-split-3857.zip',  # noqa
+        'directory': 'water-polygons-split-3857',
+        'url': 'http://osmdata.openstreetmap.de/download/water-polygons-split-3857.zip',  # noqa
```
Comment (imagico) [inline scripts/get-shapefiles.py]: ditto.

File: shapefiles.mss [hunk 3]
```diff
@@ -11,22 +11,9 @@
...
-      line-offset: 0.5;
-    }
-  }
-}
-
-#coast-poly {
-  [zoom >= 10] {
-    polygon-fill: @land-color;
+.ocean {
+  [zoom >= 0] {
```
Comment (imagico) [inline shapefiles.mss]: Don't think that is really necessary.

**Comment by jeisenbe** [comment]
I’ve pushed the requested changes and tested that the rendering is correct.
It should be ready to merge.

**Comment by jeisenbe** [comment]
I downloaded the new shapefiles from [osmdata.openstreetmap.de](https://osmdata.openstreetmap.de/data/water-polygons.html) and checked that the rendering is still correct. I will submit a new PR to suggest using this source, once this PR is merged.

I believe this PR is ready to merge now. Are there any concerns?

**Comment by imagico** [comment]
I have merged this now so we can move forward with the various matters that depend on ocean polygons.

It seems we have consensus that reverting #3065 at least an acceptable side effect of this change - if not desirable on its own because of the problems discussed in #3695.  If i misread anyone on this please say so.

As said we should move to osmdata.openstreetmap.de as shapefile data source with the same release as this change to avoid style users having to switch data source and deal with potential issues twice.

**Comment by matkoniecz** [comment]
> It seems we have consensus that reverting #3065 at least an acceptable side effect of this change

I agree now.

**Comment by kocio-pl** [comment]
Although I think it's better to have coastline simplification, this patch has higher priority for me, so I'm happy to see it merged - thanks.

**Pull request merged**